packages/dht/src/connection/WebSocket/WebSocketServer.ts

```diff
@@ -0,0 +1,123 @@
+                    logger.info((new Date()) + ' Server is listening on port ' + port)
+                    resolve()
+                })
+            }
+
+            else if (port) {
+                this.httpServer.listen(port, () => {
+                    logger.info((new Date()) + ' Server is listening on port ' + port)
+                    resolve()
+                })
+            }
+
+            else {
+                reject('Listen port for WebSocket server not given')
+            }
+
+            if (typeof NodeJsWsServer !== 'undefined') {
+                this.wsServer = new NodeJsWsServer({
+                    httpServer: this.httpServer,
+                    autoAcceptConnections: false
+                })
+            }
+            else {
+                this.wsServer = new WsServer({
+                    httpServer: this.httpServer,
+                    // You should not use autoAcceptConnections for production
+                    // applications, as it defeats all standard cross-origin protection
+                    // facilities built into the protocol and the browser.  You should
+                    // *always* verify the connection's origin and decide whether or not
+                    // to accept it.
+                    autoAcceptConnections: false
+                })
+            }
+
+            function originIsAllowed(_uorigin: string) {
+                // put logic here to detect whether the specified origin is allowed.
+                return true
+            }
+
+            this.wsServer.on('request', (request) => {
+                if (!originIsAllowed(request.origin)) {
+                    // Make sure we only accept requests from an allowed origin
+                    request.reject()
+                    logger.trace((new Date()) + ' IConnection from origin ' + request.origin + ' rejected.')
+                    return
+                }
+
+                const connection = request.accept(undefined, request.origin)
+
+                logger.trace((new Date()) + ' IConnection accepted.')
+
+                this.emit(ConnectionSourceEvent.CONNECTED, new ServerWebSocket(connection))
+            })
+        })
+    }
+
+    bindListeners(connectivityRequestHandler: TODO, incomingMessageHandler: TODO): void {
+        this.on(ConnectionSourceEvent.CONNECTED, (connection: IConnection) => {
+            logger.trace('server received new connection')
+
+            connection.on(ConnectionEvents.DATA, async (data: Uint8Array) => {
+                logger.trace('server received data')
+                const message = Message.fromBinary(data)
+
+                if (message.messageType === MessageType.CONNECTIVITY_REQUEST) {
+                    logger.trace('received connectivity request')
+                    connectivityRequestHandler(connection, ConnectivityRequestMessage.fromBinary(message.body))
+                }
+
+                else if (this.ownPeerDescriptor) {
```

Comment:
Maybe some sort of initialized check and state?
```suggestion
                else if (this.isInitialized()) {
```

Also maybe instead of if-else have only else and put this check as a separate guard within that else block.

Reply:
Resolved in commit 21df700bf